### PR TITLE
57 fetch metabolism estimates

### LINKED ISSUE #57: gather metabolism data
To explicitly represent metabolism in the model (i.e., to do #56) we will need to gather metabolism data: GPP, ER, and K. Metabolism estimates for 6 sites in our study area (7 if we add back 01481500 (see #50)) are included in the[ Appling et. al data release](https://www.sciencebase.gov/catalog/item/59bff507e4b091459a5e0982). Included in those are our 2 validation sites.

image

### PR DESCRIPTION
**Pull request by jsadler2**
Fetches metabolism estimates and matches to sites.

closes #57

### CODE REVIEW

**Comment by lekoenig** [comment]
Does it make sense to merge the changes in main into this branch given that some of the code changes are now redundant (e.g. in 2a_model/src/model_ready_data_utils.R)?

File: 1_fetch.R [hunk 1]
```diff
@@ -186,6 +186,27 @@ p1_targets_list <- list(
...
+    unzip(zipfile=metab_file, exdir = dirname(metab_file), overwrite=TRUE)
+    file.path(dirname(metab_file), "daily_predictions.tsv")
+    },
+    format="file" 
+  ),
+  
+  # Download and unzip metabolism estimates
+  tar_target(
+    p1_metab,
+    {
```
Comment (lekoenig) [inline 1_fetch.R:207]: I don't think the curly braces are needed here to read in the tsv file
```suggestion
    
```

File: 1_fetch.R [hunk 2]
```diff
@@ -186,6 +186,27 @@ p1_targets_list <- list(
...
+    metab_file <- download_sb_file(sb_id = "59eb9c0ae4b0026a55ffe389",
+                                   file_name = "daily_predictions.zip",
+                                   out_dir="1_fetch/out")
+    unzip(zipfile=metab_file, exdir = dirname(metab_file), overwrite=TRUE)
+    file.path(dirname(metab_file), "daily_predictions.tsv")
+    },
+    format="file" 
+  ),
+  
+  # Download and unzip metabolism estimates
```
Comment (lekoenig) [inline 1_fetch.R:204]:
```suggestion
  # Download and unzip metabolism estimates from https://www.sciencebase.gov/catalog/item/59eb9c0ae4b0026a55ffe389
```

File: 1_fetch.R [hunk 3]
```diff
@@ -186,6 +186,27 @@ p1_targets_list <- list(
...
+  tar_target(
+    p1_metab,
+    {
+      read_tsv(p1_metab_tsv, show_col_types = FALSE)
```
Comment (lekoenig) [inline 1_fetch.R:208]: Adding the renaming step here would eliminate the need to repeat this step twice (in `p2a_trn_do_zarr` and `p2a_trn_val_do_zarr`). If we use the metabolism estimates elsewhere in the pipeline down the road, it'd also be helpful to have the site names match our format. 

```suggestion
      read_tsv(p1_metab_tsv, show_col_types = FALSE) %>%
      # create a new column "site_id". This column is the same as site_name from the
      # original data, but the 'nwis_' before the site number is removed to match site naming
      # conventions used in our pipeline.
      mutate(site_id = str_replace(site_name, "nwis_", ""))
```

File: 2a_model.R [hunk 4]
```diff
@@ -79,17 +79,31 @@ p2a_targets_list <- list(
...
 
-  # write trn do data to zarr
+  # write trn do and metab data to zarr
   tar_target(
     p2a_trn_do_zarr,
-    subset_and_write_zarr(p2_daily_with_seg_ids, "2a_model/out/well_observed_trn_do.zarr", p2a_trn_sites),
+    {
+      # need to join the metab data with the DO observations. First we create a new column "site_id"
+      # (to match the p2_daily_with_seg_ids column name). This column is the same as site_name of p1_metab,
+      # but the 'nwis_' before the site number is removed so that the join can be made.
```
Comment (lekoenig) [inline 2a_model.R:88]: If you think it makes sense to move the renaming step to `p1_metab`

```suggestion
      # need to join the metab data with the DO observations. 
```

File: 2a_model.R [hunk 5]
```diff
@@ -79,17 +79,31 @@ p2a_targets_list <- list(
...
+      do_and_metab <- p2_daily_with_seg_ids %>%
+          left_join(p1_metab %>% mutate(site_id = str_replace(site_name, "nwis_", "")))
```
Comment (lekoenig) [inline 2a_model.R:90]: If you want to omit the message `Joining, by = c("site_id", "date")` that is printed when building this target, you could make the key columns explicit, e.g.:

```
p2_daily_with_seg_ids %>%
    left_join(p1_metab, by = c("site_id","date"))
```

File: 2a_model.R [hunk 6]
```diff
@@ -79,17 +79,31 @@ p2a_targets_list <- list(
...
     p2a_trn_val_do_zarr,
-    subset_and_write_zarr(p2_daily_with_seg_ids, "2a_model/out/well_observed_trn_do.zarr", p2a_trn_val_sites),
+    {
+      # need to join the metab data with the DO observations. First we create a new column "site_id"
+      # (to match the p2_daily_with_seg_ids column name). This column is the same as site_name of p1_metab,
+      # but the 'nwis_' before the site number is removed so that the join can be made.
+      do_and_metab <- p2_daily_with_seg_ids %>%
+          left_join(p1_metab %>% mutate(site_id = str_replace(site_name, "nwis_", "")))
+      subset_and_write_zarr(do_and_metab, "2a_model/out/well_observed_trn_val_targets.zarr", p2a_trn_val_sites)
+    },
```
Comment (lekoenig) [inline 2a_model.R:106]: Same comments as above for `p2a_trn_do_zarr`

**Review by lekoenig — APPROVED** [review]
Looks good, Jeff! I made a few minor comments and suggestions for your consideration but I'm not requesting any changes. Feel free to merge whenever you're ready.

File: 2a_model.R [hunk 7]
```diff
@@ -79,17 +79,31 @@ p2a_targets_list <- list(
...
+  # write trn and val do and metab data to zarr
   tar_target(
     p2a_trn_val_do_zarr,
-    subset_and_write_zarr(p2_daily_with_seg_ids, "2a_model/out/well_observed_trn_do.zarr", p2a_trn_val_sites),
+    {
+      # need to join the metab data with the DO observations. First we create a new column "site_id"
+      # (to match the p2_daily_with_seg_ids column name). This column is the same as site_name of p1_metab,
+      # but the 'nwis_' before the site number is removed so that the join can be made.
+      do_and_metab <- p2_daily_with_seg_ids %>%
+          left_join(p1_metab %>% mutate(site_id = str_replace(site_name, "nwis_", "")))
```
Comment (jsadler2) [inline 2a_model.R:104]:
```suggestion
    {
      # need to join the metab data with the DO observations. 
      do_and_metab <- p2_daily_with_seg_ids %>%
          left_join(p1_metab, by=c("site_id", "date"))
```

File: 1_fetch.R (hunk 3 above)
Comment (jsadler2) [reply 1_fetch.R]: Great idea.

**Pull request merged**